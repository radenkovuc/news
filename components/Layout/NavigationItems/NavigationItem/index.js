import styled from 'styled-components'
import React from 'react'
import withTranslation from '../../../HOCs/withTranslation'
import {Link} from '../../../../i18n'

const Item = styled.div`
  display: flex;
  border-right: 1px solid #444;
  justify-content: center;
  background: ${(props) => (props.isActive ? '#777777' : '#ffffff')};
  color: ${(props) => (props.isActive ? '#ffffff' : '#000000')};
  cursor: pointer;
  flex: 1;

  @media only screen and (min-width: 600px) {
    flex: none;
  }
`
const Text = styled.span`
  padding: 20px;
  font-size: 20px;
  font-family: 'Nunito Sans', sans-serif;
`

const NavigationItem = (props) => {
  const {text, link, isActive, t} = props

  return (
    <Item isActive={isActive}>
      <Link href={link}>
        <Text>{t(text)}</Text>
      </Link>
    </Item>
  )
}

export default withTranslation(NavigationItem)
