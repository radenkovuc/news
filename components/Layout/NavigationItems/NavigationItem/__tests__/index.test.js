import React from 'react'
import renderer from 'react-test-renderer'
import NavigationItem from '../index'

describe('<NavigationItem />', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<NavigationItem />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
