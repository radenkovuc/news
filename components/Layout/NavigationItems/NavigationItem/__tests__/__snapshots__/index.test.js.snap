// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationItem /> match snapshot 1`] = `
<div
  className="Item-sc-14w5znm-0 SlJzJ"
>
  <span
    className="Text-sc-14w5znm-1 jGVkFb"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    
  </span>
</div>
`;
